src/web/mjs/connectors/ComicBoost.mjs

```diff
@@ -13,32 +13,169 @@ export default class ComicBoost extends Publus {
 
     async _getMangaFromURI(uri) {
         let request = new Request(uri, this.requestOptions);
-        let data = await this.fetchDOM(request, 'article#book div.detail h1.name');
+        let data = await this.fetchDOM(request, 'h1.comic-title');
         let id = uri.pathname + uri.search;
         let title = data[0].textContent.trim();
         return new Manga(this, id, title);
     }
 
     async _getMangas() {
-        let request = new Request(new URL('/series?per=1000', this.url), this.requestOptions );
-        let data = await this.fetchDOM(request, 'div.box ul li div.info p.name a');
+        let mangaList = [];
+        const uri = new URL('/genre/', this.url);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, 'ul.pagination-list.right li.to-last a');
+        const pageCount = parseInt(data[0].href.match(/(\d+)$/)[1]);
+        for(let page = 1; page <= pageCount; page++) {
+            const mangas = await this._getMangasFromPage(page);
+            mangaList.push(...mangas);
+        }
+        return mangaList;
+    }
+
+    async _getMangasFromPage(page) {
+        const uri = new URL('/genre/?p=' + page, this.url);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, 'div.book-list-item-inner');
         return data.map(element => {
             return {
-                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
-                title: element.text.trim()
+                id: this.getRootRelativeOrAbsoluteLink(element.querySelector('a.book-list-item-thum-wrapper'), this.url),
+                title: element.querySelector('source.thum').getAttribute('title').trim()
             };
         });
     }
 
     async _getChapters(manga) {
         let request = new Request(new URL(manga.id, this.url), this.requestOptions );
-        let data = await this.fetchDOM(request, 'section#productList ul.list_item--series li div.product_item ul.box_button li a.button');
+        let data = await this.fetchDOM(request, 'a.book-product-list-item');
         return data.map(element => {
             return {
                 id: this.getRootRelativeOrAbsoluteLink(element, this.url),
-                title: element.closest('div.product_item').querySelector('p.name').textContent.trim(),
+                title: element.dataset.title.trim(),
                 language: ''
             };
-        });
+        }).filter(el => !el.id.includes('?coin=')); //exclude not accessible chapters
+    }
+
+    async _getPages(chapter) {
+
+        const script = `
+            new Promise((resolve, reject) => {
+                setTimeout(async () => {
+
```

Comment:
This method looks quite big, recommend to wrap this whole method body into a `try {...} catch(error) {  reject(error); }` block statement to ensure the promise correctly delivers an error (otherwise the error is ignored and the promise never finishes correctly)